Add warning note in RabbitMQ package readme about possible discrepancy after RabbitMQ upgrade (#15101)
elastic/integrations: packages/rabbitmq/_dev/build/docs/README.md

## Compatibility

> The RabbitMQ integration is fully tested with RabbitMQ 3.7.4 and 4.1.3, and it should be compatible with any version supporting the management plugin (which needs to be installed and enabled). It has also been tested with 3.6.0, 3.6.5, and 3.7.14.

Reviewer: It has also been tested with 3.6.0, 3.6.5, and 3.7.14.

Why do we need this separately ?

Reviewer: @giorgi-imerlishvili-elastic You tried it with 3.7.14 and 4.1.3.
Is that right ?

Author: > @giorgi-imerlishvili-elastic You tried it with 3.7.14 and 4.1.3. Is that right ?

yes right.

> It has also been tested with 3.6.0, 3.6.5, and 3.7.14.
> 
> Why do we need this separately ?

Previously it was written like this: "Exchange dataset is also been tested with 3.6.0, 3.6.5, and 3.7.14."
 I removed this part "Exchange dataset is" that's why I think now it looks little bit confusing

Author: ```suggestion
The RabbitMQ integration is fully tested with RabbitMQ 3.7.4 and 4.1.3, and it should be compatible with any version supporting the management plugin (which needs to be installed and enabled). Exchange dataset is also tested with 3.6.0, 3.6.5 and 3.7.14.
```
I think will be better if i bring back how it was written before, I don't know why `Exchange dataset` were mentioned separately that's why I removed it but looks like removing it causes more confusion.

> The RabbitMQ integration is fully tested with RabbitMQ 3.7.4 and 4.1.3, and it should be compatible with any version supporting the management plugin (which needs to be installed and enabled). It has also been tested with 3.6.0, 3.6.5, and 3.7.14.

Reviewer 2: ```suggestion
The RabbitMQ integration is fully tested with RabbitMQ versions 3.7.4 and 4.1.3, and it should be compatible with any version supporting the management plugin, which must be installed and enabled. It has also been tested with 3.6.0, 3.6.5, and 3.7.14.
```
Sounds like this info is relevant, might not deserve brackets...

The application logs dataset parses single file format introduced in 3.7.0.

> When upgrading RabbitMQ to version 4.1.3 or above, the `/api/nodes` endpoint no longer returns the `rabbitmq.node.queue.index.journal_write` metric in its response. This metric is used in the Kibana dashboard "Queue Index Operations [Metrics RabbitMQ]". As a result, after upgrading to 4.1.3 or later, this metric and related visualizations in the dashboard will be missing or incomplete.

Reviewer 3: this endpoint is the RabbitMQ endpoint, right? Maybe specify that so there's no confusion that it's an Elastic endpoint's fault

> When upgrading RabbitMQ to version 4.1.3 or above, the `/api/nodes` endpoint (from RabbitMQ's management API) no longer returns the `rabbitmq.node.queue.index.journal_write` metric in its response. This metric is used in the Kibana dashboard "Queue Index Operations [Metrics RabbitMQ]". As a result, after upgrading to 4.1.3 or later, this metric and related visualizations in the dashboard will be missing or incomplete.

Reviewer 2: ```suggestion
When upgrading RabbitMQ to version 4.1.3 or higher, the `/api/nodes` endpoint (from RabbitMQ's management API) no longer returns the `rabbitmq.node.queue.index.journal_write` metric in its response. This metric is used in the Kibana dashboard "Queue Index Operations [Metrics RabbitMQ]". As a result, after upgrading to 4.1.3 or higher, this metric and related visualizations in the dashboard will be missing or incomplete.
```
